cpp/src/arrow/csv/options.h

```diff
@@ -194,6 +194,9 @@ struct ARROW_EXPORT WriteOptions {
   /// This number can impact performance.
   int32_t batch_size = 1024;
 
+  /// Field delimiter
+  char delimiter = ',';
```

Comment:
Shall we treat the customized delimiter as non structural chars in non-quoted string checking [1]? Currently, it's hard coded to check `,` and `\r`, `\n`, `"`.

https://github.com/apache/arrow/blob/master/cpp/src/arrow/csv/writer.cc#L213

Reply:
I am not sure about its implementation, could you please help me with it.

cc: @westonpace @lidavidm

Reply:
No worry, I will upload the change above your PR.

Reply:
Thanks!

Reply:
[to-sanjiban.diff.gz](https://github.com/apache/arrow/files/8094142/to-sanjiban.diff.gz)
@sanjibansg , you can apply above diff to your code. Please note this is a quick fix, needs adding test cases, or refinement maybe, and may have format issue.

Reply:
okay thanks applying this diff. Thanks a lot @cyb70289!